test/test_triplets_classifiers.py

```diff
@@ -63,3 +64,48 @@ def test_accuracy_toy_example(estimator, build_dataset):
   # we force the transformation to be identity so that we control what it does
   estimator.components_ = np.eye(X.shape[1])
   assert estimator.score(triplets_test) == 0.25
+
+
+@pytest.mark.parametrize('estimator, build_dataset', triplets_learners,
+                         ids=ids_triplets_learners)
+def test_no_zero_prediction(estimator, build_dataset):
+  """
+  Test that all predicted values are in {-1, 1}, even when the
+  distance d(x,y) and d(x,z) is the same for a triplet of the
+  form (x, y, z).
+  """
+  # Dummy fit
+  triplets, _, _, X = build_dataset(with_preprocessor=False)
+  # Force 3 dimentions only, to use cross product and get easy orthogonal vec.
+  triplets = np.array([[t[0][:3], t[1][:3], t[2][:3]] for t in triplets])
+  X = np.array([x[:3] for x in X])
+  # Dummy fit
+  estimator = clone(estimator)
+  set_random_state(estimator)
+  estimator.fit(triplets)
+  # we force the transformation to be identity, to force euclidean distance
+  estimator.components_ = np.eye(X.shape[1])
+
+  # Get two orthogonal vectors in respect to X[1]
+  k = X[1]/np.linalg.norm(X[1])  # Normalize first vector
+  x = X[2] - X[2].dot(k) * k  # Get random orthogonal vector
+  x /= np.linalg.norm(x)  # Normalize
+  y = np.cross(k, x)  # Get orthogonal vector to x
+  # Assert these orthogonal vectors are different
+  with pytest.raises(AssertionError):
+    assert_array_equal(X[1], x)
+  with pytest.raises(AssertionError):
+    assert_array_equal(X[1], y)
+  # Assert the distance is the same for both
+  assert estimator.get_metric()(X[1], x) == estimator.get_metric()(X[1], y)
+
+  # Form the three scenarios where predict() gives 0 with numpy.sign
+  triplets_test = np.array(  # Critical examples
+    [[X[0], X[2], X[2]],
+     [X[1], X[1], X[1]],
+     [X[1], x, y]])
+  # Predict
+  predictions = estimator.predict(triplets_test)
+  # Count non -1 or 1 values
+  not_valid = [e for e in predictions if e not in [-1, 1]]
+  assert len(not_valid) == 0
```

Comment:
this does not really match the semantics of the name of the test (which is to verify that there is no zero value). so either you just check that there is no zero value (and adapt the docstring of the test accordingly), or you can try to merge this with the already existing test `test_predict_only_one_or_minus_one`.